src/index.js

```diff
@@ -88,12 +102,29 @@ export default {
           console.warn(`Scale not found for '${axis}'-axis in datasets[${i}] of chart id ${chart.id}, skipping.`);
           continue;
         }
-        if (colors) {
+        const stateOptions = getStateOptions(state, meta, key, i);
+        if (colors && !meta.hidden) {
+          const option = {
+            datasetIndex: i,
+            axis,
+            scale,
+            stopColors: []
+          };
+          stateOptions.push(option);
           const value = createGradient(ctx, axis, scale);
-          addColors(value, scale, colors);
+          addColors(scale, colors, option.stopColors);
+          applyColors(value, option.stopColors);
           setValue(meta, dataset, key, value);
         }
       }
     }
+  },
+
+  afterUpdate(chart) {
+    const state = chartStates.get(chart);
+    if (chart.legend && Chart.version) {
```

Comment:
Because the Chart.version is starting to appear more frequently, should add a constant about it at the top:

```js
const isChartV3 = Chart.version;
```

Reply:
OK!